rcgen/Cargo.toml

```diff
@@ -38,9 +38,9 @@ zeroize = { version = "1.2", optional = true }
 [features]
 default = ["crypto", "pem", "ring"]
 crypto = []
-aws_lc_rs = ["crypto", "dep:aws-lc-rs"]
+aws_lc_rs = ["crypto", "dep:aws-lc-rs", "aws-lc-rs/aws-lc-sys"]
 ring = ["crypto", "dep:ring"]
-fips = ["aws_lc_rs", "aws-lc-rs?/fips"]
```

Comment:
I removed the `?` because it didn't make sense, or was it there for a reason?

Reply:
I think you're right that in this context it doesn't make sense and should be removed. My understanding of the `?` is that it:
> "will only enable the given feature if something else enables the optional dependency."

But in this case, the `fips` feature is activating `aws_lc_rs` right alongside `aws-lc-rs/fips`. 

@djc Does that match your understanding? The usage came along with https://github.com/rustls/rcgen/pull/234 and wasn't discussed explicitly as far as I can tell.

Reply:
IIRC there was a bug in older Cargo versions that needed the ? to be there. If it works without it on the project's MSRV, I'm fine with this.

Reply:
I tried it locally with Rust v1.66 and it compiles for me.

Reply:
`"aws_lc_rs", "aws-lc-rs?/fips"` is the new way of writing `"aws-lc-rs/fips"` and makes it explicit that the enabling the `aws-lc-rs` feature is vital for the `fips` feature.

Reply:
In this case I don't think it makes sense to preserve the `?`, since we unconditionally enable the optional dependency anyway when the `fips` feature is enabled.

Reply:
> In this case I don't think it makes sense to preserve the ?

I agree. Since @est31 didn't request any changes and approved the PR I think it's fair to merge with the removal?